### PR TITLE
chore(functions): switch back to Pub/Sub POJOs

### PR DESCRIPTION
**Pull request by ace-n**
This reverts commit 23dc8f9205375285cd68570ed10910abee3baac9 (PR #4869)

**Googlers:** fixes b/255813535

### CODE REVIEW

**Comment by ace-n** [comment]
**TODO:** add code comments w/ proto links

File: functions/concepts/retry-pubsub/src/main/java/functions/eventpojos/PubSubMessage.java [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package functions.eventpojos;
+
+// [START functions_helloworld_pubsub_message]
+// This POJO is based on the `PubsubMessage` object described here:
+// https://github.com/googleapis/googleapis/blob/master/google/pubsub/v1/pubsub.proto
```
Comment (averikitsch) [inline functions/concepts/retry-pubsub/src/main/java/functions/eventpojos/PubSubMessage.java:21]: Let's link to https://cloud.google.com/pubsub/docs/reference/rest/v1/PubsubMessage instead

File: functions/concepts/retry-timeout/src/main/java/functions/eventpojos/PubSubMessage.java [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+package functions.eventpojos;
+
+// [START functions_helloworld_pubsub_message]
+// This POJO is based on the `PubsubMessage` object described here:
+// https://github.com/googleapis/googleapis/blob/master/google/pubsub/v1/pubsub.proto
+
+import java.util.Map;
+
+public class PubSubMessage {
+  // Cloud Functions uses GSON to populate this object.
```
Comment (averikitsch) [inline functions/concepts/retry-timeout/src/main/java/functions/eventpojos/PubSubMessage.java:26]:
```suggestion
  // Cloud Functions uses Gson to deserialize JSON into a Java object
```

File: functions/concepts/retry-timeout/src/main/java/functions/eventpojos/PubSubMessage.java [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+  // Field types/names are specified by Cloud Functions
+  // Changing them may break your code!
+  private String data;
+  private Map<String, String> attributes;
+  private String messageId;
+  private String publishTime;
+
```
Comment (averikitsch) [inline functions/concepts/retry-timeout/src/main/java/functions/eventpojos/PubSubMessage.java:33]: Should we also be providing a field for "orderingKey"?

File: functions/pubsub/subscribe-to-topic/src/main/java/functions/eventpojos/PubSubMessage.java [hunk 4]
```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright 2020 Google LLC
```
Comment (averikitsch) [inline functions/pubsub/subscribe-to-topic/src/main/java/functions/eventpojos/PubSubMessage.java:2]:
```suggestion
 * Copyright 2022 Google LLC
```

File: functions/ocr/ocr-translate-text/src/main/java/functions/eventpojos/PubSubMessage.java [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+
+package functions.eventpojos;
+
+// [START functions_helloworld_pubsub_message]
+// This POJO is based on the `PubsubMessage` object described here:
+// https://github.com/googleapis/googleapis/blob/master/google/pubsub/v1/pubsub.proto
+
+import java.util.Map;
+
+public class PubSubMessage {
```
Comment (averikitsch) [inline functions/ocr/ocr-translate-text/src/main/java/functions/eventpojos/PubSubMessage.java:25]:
```suggestion
public class PubsubMessage {
```
To match other documentation and protos

**Review by averikitsch — CHANGES_REQUESTED** [review]

File: functions/pubsub/subscribe-to-topic/src/main/java/functions/eventpojos/PubSubMessage.java (hunk 4 above)
Comment (ace-n) [reply functions/pubsub/subscribe-to-topic/src/main/java/functions/eventpojos/PubSubMessage.java:2]: Done.

File: functions/concepts/retry-timeout/src/main/java/functions/eventpojos/PubSubMessage.java (hunk 2 above)
Comment (ace-n) [reply functions/concepts/retry-timeout/src/main/java/functions/eventpojos/PubSubMessage.java:26]: Done.

File: functions/concepts/retry-pubsub/src/main/java/functions/eventpojos/PubSubMessage.java (hunk 1 above)
Comment (ace-n) [reply functions/concepts/retry-pubsub/src/main/java/functions/eventpojos/PubSubMessage.java:21]: Done.

File: functions/ocr/ocr-translate-text/src/main/java/functions/eventpojos/PubSubMessage.java (hunk 5 above)
Comment (ace-n) [reply functions/ocr/ocr-translate-text/src/main/java/functions/eventpojos/PubSubMessage.java:25]: Done.

File: functions/concepts/retry-timeout/src/main/java/functions/eventpojos/PubSubMessage.java (hunk 3 above)
Comment (ace-n) [reply functions/concepts/retry-timeout/src/main/java/functions/eventpojos/PubSubMessage.java:33]: I'd argue **no**, as Cloud Functions-based Pub/Sub topics [don't support ordered messages](https://stackoverflow.com/questions/65692307/how-to-deploy-pubsub-triggered-cloud-function-with-message-ordering#comment116183324_65694250).

File: functions/concepts/retry-pubsub/src/main/java/functions/eventpojos/PubsubMessage.java [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package functions.eventpojos;
+
+// [START functions_helloworld_pubsub_message]
```
Comment (ace-n) [inline functions/concepts/retry-pubsub/src/main/java/functions/eventpojos/PubsubMessage.java:19]: Do we need these region tags?

Comment (averikitsch) [reply functions/concepts/retry-pubsub/src/main/java/functions/eventpojos/PubsubMessage.java:19]: No let's remove

File: functions/concepts/retry-timeout/src/main/java/functions/eventpojos/PubSubMessage.java (hunk 3 above)
Comment (averikitsch) [reply functions/concepts/retry-timeout/src/main/java/functions/eventpojos/PubSubMessage.java]: sgtm

**Review by averikitsch — APPROVED** [review]
Please address errors:
[ERROR] /workspace/src/test/java/functions/ScalaHelloPubSubTest.java:[23,28] cannot find symbol
[ERROR]   symbol:   class PubsubMessage
[ERROR]   location: package functions.eventpojos
[ERROR] /workspace/src/test/java/functions/ScalaHelloPubSubTest.java:[57,5] cannot find symbol
[ERROR]   symbol:   class PubsubMessage
[ERROR]   location: class functions.ScalaHelloPubSubTest
[ERROR] /workspace/src/test/java/functions/ScalaHelloPubSubTest.java:[57,33] cannot find symbol
[ERROR]   symbol:   class PubsubMessage
[ERROR]   location: class functions.ScalaHelloPubSubTest
[ERROR] /workspace/src/test/java/functions/ScalaHelloPubSubTest.java:[70,39] cannot find symbol
[ERROR]   symbol:   class PubsubMessage
[ERROR]   location: class functions.ScalaHelloPubSubTest

File: functions/concepts/retry-pubsub/src/main/java/functions/eventpojos/PubsubMessage.java (hunk 6 above)
Comment (ace-n) [reply functions/concepts/retry-pubsub/src/main/java/functions/eventpojos/PubsubMessage.java]: Done.

**Pull request merged**